chia/util/streamable.py

```diff
@@ -295,7 +295,7 @@ def function_to_parse_one_item(cls: Type[cls.__name__], f_type: Type):  # type:
         raise NotImplementedError(f"Type {f_type} does not have parse")
 
     @classmethod
-    def parse(cls: Type[cls.__name__], f: BinaryIO) -> cls.__name__:  # type: ignore
+    def parse(cls: Type[Streamable], f: BinaryIO) -> Streamable:
```

Comment:
Since we inherit, this is not accurate.  The type of `cls` won't be `Streamable` specifically when accessed on an inherited class.  Rather we need something like `_T_Streamable = TypeVar("_T_Streamable", bound="Streamable")` and then:

```suggestion
    def parse(cls: Type[_T_Streamable], f: BinaryIO) -> _T_Streamable:
```

This says that the method will return an instance of whatever `cls` is.